jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/PushTagToRepositoryAction.kt

```diff
@@ -150,20 +209,155 @@ internal class PushToEcrDialog(
                     text = value.tag ?: value.imageId.take(15)
                 }
             )
-                .sizeGroup(selectorGroup)
+                .constraints(grow)
                 .withErrorOnApplyIf(message("ecr.image.not_selected")) { it.selected() == null }
         }
+    }
 
-        row(message("ecr.push.remoteTag")) {
-            textField(::remoteTag)
-                .sizeGroup(selectorGroup)
-                .withErrorOnApplyIf(message("ecr.tag.not_provided")) { it.blankAsNull() == null }
+    private fun dockerfileConfigurationSelectorPanel() = panel {
+        row(message("ecr.dockerfile.configuration.label")) {
+            cell {
+                val model = CollectionComboBoxModel<DockerRunConfiguration>()
+                rebuildRunConfigurationComboBoxModel(model)
+                val box = comboBox(
+                    model,
+                    { runConfiguration },
+                    { ::runConfiguration.set(it) },
+                    listCellRenderer { value, _, _ ->
+                        icon = value.icon
+                        text = value.name
+                    }
+                )
+                    .constraints(grow)
+                    .withErrorOnApplyIf(message("ecr.dockerfile.configuration.invalid")) { it.selected() == null }
+                    .withErrorOnApplyIf(message("ecr.dockerfile.configuration.invalid_server")) { it.selected()?.serverName == null }
+
+                // TODO: how do we render both the Docker and action items correctly?
+                box.component.apply {
+                    isEditable = true
+                    editor = object : BasicComboBoxEditor.UIResource() {
+                        override fun createEditorComponent(): JTextField {
+                            val textField = ExtendableTextField()
+                            textField.isEditable = false
+
+                            buildDockerfileActions(model, textField)
+                            textField.border = null
+
+                            return textField
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun buildDockerfileActions(runConfigModel: CollectionComboBoxModel<DockerRunConfiguration>, textComponent: ExtendableTextField) {
+        val editExtension = ExtendableTextComponent.Extension.create(
+            AllIcons.General.Inline_edit, AllIcons.General.Inline_edit_hovered,
+            message("ecr.dockerfile.configuration.edit")
+        ) {
+            runConfiguration?.let {
+                RunManager.getInstance(project).findSettings(it)?.let { settings ->
```

Comment:
We can set a max size so the panel isn't so wide:
```maximumSize = Dimension(400, 32767)```

Reply:
opted for `growPolicy(GrowPolicy.MEDIUM_TEXT)` instead

Reply:
That works better, cool